app/experimenter/static/js/design-page.js

```diff
@@ -0,0 +1,494 @@
+      <Row>
+        <Col md={3} className="text-right mb-3">
+          <FormLabel>
+            <strong>Active Experiment Name</strong>
+          </FormLabel>
+          <br />
+          <a href="/">help</a>
+        </Col>
+        <Col md={9}>
+          <FormControl
+            type="text"
+            name="addon-experiment-id"
+            onChange={props.handleInputChange}
+            value={props.addon_experiment_id}
+          />
+        </Col>
+      </Row>
+      <Row>
+        <Col md={3} className="text-right mb-3">
+          <FormLabel>
+            <strong>Signed Release URL</strong>
+          </FormLabel>
+          <br />
+          <a href="/">help</a>
+        </Col>
+        <Col md={9}>
+          <FormControl
+            type="text"
+            name="addon-release-url"
+            onChange={props.handleInputChange}
+            value={props.addon_release_url}
+          />
+        </Col>
+      </Row>
+    </div>
+  );
+}
+
+function GenericForm(props) {
+  return (
+    <div>
+      <Row>
+        <Col md={{ span: 4, offset: 3 }}>
+          <h4>Generic</h4>
+        </Col>
+      </Row>
+      <Row>
+        <Col md={3} className="text-right mb-3">
+          <FormLabel>
+            <strong>Design</strong>
+          </FormLabel>
+          <br />
+          <a href="/">help</a>
+        </Col>
+        <Col md={9}>
+          <FormControl
+            name="design"
+            type="textarea"
+            value={props.design}
+            onChange={props.handleInputChange}
+          />
+        </Col>
+      </Row>
+    </div>
+  );
+}
+
+function TypeForm(props) {
+  const type = props.type;
+
+  if (type == "pref") {
+    return <PrefForm {...props}/>;
+  } else if (type == "addon") {
+    return <AddonForm {...props}/>;
+  } else {
+    return <GenericForm {...props}/>;
+  }
+}
+
+export default class DesignForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.slug = props.slug
+
+    this.addBranch = this.addBranch.bind(this);
+    this.removeBranch = this.removeBranch.bind(this);
+
+    this.updateRatio = this.updateRatio.bind(this);
+    this.updateName = this.updateName.bind(this);
+    this.updateDescription = this.updateDescription.bind(this);
+    this.updateValue = this.updateValue.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleInputChange = this.handleInputChange.bind(this);
+
+    this.state = {
+      type: props.expType,
+      pref_key: "",
+      pref_type: "",
+      pref_branch: "",
+      addon_experiment_id: "",
+      addon_release_url: "",
+      design: "",
+      variants: [{
+        ratio: null,
+        name: "",
+        description: "",
+        value: "",
+        is_control: true,
+      }, {
+        ratio: null,
+        name: "",
+        description: "",
+        value: "",
+        is_control: false,
+      }],
+      errors: {}
+    };
+  }
+
+  componentDidMount() {
+    const that = this;
```

Comment:
I'm not a JS expert but I think if you use arrow functions `(arg) => { body }` then it uses `this` from the parent scope, and then you don't need this alias? @rehandalal ?

Reply:
good catch, yes arrow functions do not have their own binding to `this` so it ends up using `this` from parent scope. (see: [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions))

Reply:
uh, I just removed the `this = that` thing and all was fine so... 🤷‍♀